src/sdk/namespace_fs.js

```diff
@@ -79,6 +79,22 @@ function sort_entries_by_name(a, b) {
     return 0;
 }
 
+function _get_mtime_from_filename(filename) {
+    const file_parts = filename.split('-');
+    return parseInt(file_parts[1], 36);
+}
+
+/**
+ * @param {fs.Dirent} a
+ * @param {fs.Dirent} b
+ * @returns {Number}
+ */
+function sort_entries_by_time(a, b) {
+    const a_mtime = _get_mtime_from_filename(a.name);
+    const b_mtime = _get_mtime_from_filename(b.name);
+    return b_mtime - a_mtime;
+}
```

Comment:
you should compare first by file name and only if filenames match then compare by mtime. something like that:

```suggestion
function sort_entries_by_name_and_time(a, b) {
    const a_name = _get_filename(a.name) // will cut the version suffix
    const b_name = _get_filename(b.name)
    if (a_name === b_name) {
        const a_mtime = _get_mtime_from_filename(a.name);
        const b_mtime = _get_mtime_from_filename(b.name);
        return b_mtime - a_mtime;
    } else {
        if (a_name < b_name) return -1;
        if (a_name > b_name) return 1;
        return 0;
    }
}
```

Reply:
Yes. The latest object (not under `.versions`) wont have version suffix. So When we sort based on time, the latest object will go last in the order

Reply:
For latest object, I will always return time.now(). Just to keep it on top of the entries